data/impl/RecordSet.js

```diff
@@ -135,19 +158,32 @@ export class RecordSet {
         return new RecordSet(this.store, existingRecords);
     }
 
+    addData(rawData, parentId) {
+        const {records} = this,
+            parent = records.get(parentId),
+            newRecords = this.createRecords(rawData, parent);
+
+        newRecords.forEach(rec => {
+            throwIf(records.has(rec.id),
+                `A Record with ID ${rec.id} already exists in the RecordSet.`);
+        });
+
+        return new RecordSet(this.store, new Map([...records, ...newRecords]));
+    }
+
     removeRecords(ids) {
         const removes = new Set();
         ids.forEach(id => this.gatherDescendants(id, removes));
-        return this.applyFilter(r => !removes.has(r.id));
+        return this.applyFilter({fn: r => !removes.has(r.id)});
```

Comment:
This fixes a bug introduced with the (unreleased) StoreFilter changes, remove records was not working because it was not conforming to the new signature of `applyFilter`